x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -103,6 +105,31 @@ export function trainedModelsRoutes(
           // model_type is missing
           // @ts-ignore
           const result = resp.trained_model_configs as TrainedModelConfigResponse[];
+
+          try {
+            // Check if model is used by an inference service
+            const { models } = await client.asCurrentUser.transport.request<{
```

Comment:
As discussed briefly before, there may be a situation where a user does not have permission to list the inference services, but does have permission to delete or stop model deployments.
In that case, this list would be empty and they would not be warned.

If this `_inference/_all` request fails, It would probably be safer to return something which flags up that the user cannot list the services and therefore they should not be allowed to stop or delete the deployment.

Reply:
but that can also be misleading, they might not have any inference servies whatsoever and we disable the stop deployment actions for them 🤔 WDYT @davidkyle?

Reply:
My original thought was to run the request with the internal user and return a flag along with the list which is `hasInferenceServices`.
This would solve that problem for viewer users, and admin users would also get the full list.
The internal user request would only be run if the current user one fails due to permissions.

But this will need the internal user's permissions to be updated to allow then to call this endpoint.